packages/esm-patient-orders-app/src/lab-results/lab-results-form.component.tsx

```diff
@@ -0,0 +1,194 @@
+    setIsEditing(true);
+  }
+
+  if (isEditing && !obsUuid) {
+    setIsLoadingInitialValues(true);
+    fetchObservation(obsUuid).then((data) => {
+      if (data) {
+        setInitialValues(data);
+      }
+      setIsLoadingInitialValues(false);
+    });
+  }
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+  }, [isDirty]);
+
+  if (isLoadingConcepts) {
+    return <InlineLoading iconDescription="Loading" description="Loading test details" status="active" />;
+  }
+
+  const saveLabResults = (data, e) => {
+    setIsSubmitting(true);
+    e.preventDefault();
+    // assign result to test order
+    const documentedValues = getValues();
+    let obsValue = [];
+
+    if (concept.set && concept.setMembers.length > 0) {
+      let groupMembers = [];
+      concept.setMembers.forEach((member) => {
+        let value;
+        if (member.datatype.display === 'Numeric' || member.datatype.display === 'Text') {
+          value = getValues()[`${member.uuid}`];
+        } else if (member.datatype.display === 'Coded') {
+          value = {
+            uuid: getValues()[`${member.uuid}`],
+          };
+        }
+        const groupMember = {
+          concept: { uuid: member.uuid },
+          value: value,
+          status: 'FINAL',
+          order: { uuid: order.uuid },
+        };
+        groupMembers.push(groupMember);
+      });
+
+      obsValue.push({
+        concept: { uuid: order.concept.uuid },
+        status: 'FINAL',
+        order: { uuid: order.uuid },
+        groupMembers: groupMembers,
+      });
+    } else if (!concept.set && concept.setMembers.length === 0) {
+      let value;
+      if (concept.datatype.display === 'Numeric' || concept.datatype.display === 'Text') {
+        value = getValues()[`${concept.uuid}`];
+      } else if (concept.datatype.display === 'Coded') {
+        value = {
+          uuid: getValues()[`${concept.uuid}`],
+        };
+      }
+
+      obsValue.push({
+        concept: { uuid: order.concept.uuid },
+        status: 'FINAL',
+        order: { uuid: order.uuid },
+        value: value,
+      });
+    }
+
+    const obsPayload = { obs: obsValue };
+
+    const resultsStatusPayload = {
+      fulfillerStatus: 'COMPLETED',
+      fulfillerComment: 'Test Results Entered',
+    };
+
+    updateOrderResult(order.uuid, order.encounter.uuid, obsUuid, obsPayload, resultsStatusPayload).then(
+      () => {
+        setIsSubmitting(false);
+        closeWorkspaceWithSavedChanges();
+        mutate();
+
+        showSnackbar({
+          isLowContrast: true,
+          title: t('saveLabResults', 'Save lab results'),
+          kind: 'success',
+          subtitle: t('successfullySavedLabResults', 'Lab results for {{orderNumber}} have been successfully updated', {
+            orderNumber: order?.orderNumber,
+          }),
+        });
+      },
+      (err) => {
+        setIsSubmitting(false);
+        showNotification({
```

Comment:
I don't know why I'm getting this when attempting to Add lab results. Can you confirm whether this is the case for you too, @pirupius? 

![CleanShot 2024-04-05 at 12  07 17@2x](https://github.com/openmrs/openmrs-esm-patient-chart/assets/8509731/da8ce506-b870-40ea-8cb5-eeaef28968b9)

Additionally, this notification should use `showSnackbar` instead of `showNotification`.